conans/client/tools/win.py

```diff
@@ -346,8 +346,12 @@ def relevant_path(path):
 
 @contextmanager
 def vcvars(*args, **kwargs):
-    new_env = vcvars_dict(*args, **kwargs)
-    with environment_append(new_env):
+    settings = args[0]
+    if settings.get_safe("os") == "Windows":
```

Comment:
I think it shouldn't be based on the setting but in the current system.

Reply:
you mean using ``platform.system()`` ?

Reply:
yes